databroker/muggler/data.py

```diff
@@ -418,22 +421,24 @@ def bin_by_edges(self, bin_edges, anchor=None, time_labels=None,
             else:
                 raise ValueError("anchor must be 'left', 'center', 'right', "
                                  "or None")
-        return self.resample(time_labels, binning, interpolation, agg)
+        return self.resample(time_labels, binning, interpolation, agg, col_names=col_names)
 
     def resample(self, time_labels, binning, interpolation=None, agg=None,
-                 verify_integrity=True):
+                 verify_integrity=True, col_names=None):
         result = {}  # dict of DataFrames, to become one MultiIndexed DataFrame
-
+        df = self._dataframe
```

Comment:
You called `self._dataframe` in 3 different places in this function. Seemed a little unnecessary to go through the song and dance of creating the dataframe 3 times. This creates it once and then uses the same dataframe object